src/components/common/ButtonComponent.tsx

```diff
@@ -0,0 +1,23 @@
+import { MouseEventHandler } from "react";
+
+interface ButtonComponentProps {
+  className?: string;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  children: React.ReactNode;
+  disabled: boolean;
+}
+const ButtonComponent = (props: ButtonComponentProps) => {
```

Comment:
You can destruct props in the component, so you don't have to use `props.className`, `props.onClick`, etc.

```suggestion
const ButtonComponent = ({ className, onClick, children, disabled }: ButtonComponentProps) => {
```